### PR TITLE
Fix: product images not rendering in bundle previews (#73)

### LINKED ISSUE #73: Product Images Are Not Rendering in Previews
### **📌 Description**
- Product images do not render in bundle previews 

### **✅ Acceptance Criteria**

- [ ] Product images show in all previews

Image

Image

Image

### PR DESCRIPTION
**Pull request by HeyItsChloe**
## Summary

Closes #73.

Product images were not rendering in any bundle preview because several `img` components used `src={product.media}` without a fallback value or an `onError` handler. When `product.media` is `null`, `undefined`, or a broken URL the browser received an empty or invalid `src` and displayed a broken-image icon.

## Root cause

Five components rendered product images without a fallback:

| File | Previous | Fixed |
|------|----------|-------|
| `bundleDiscountActions.jsx` | `src={product.media}` | `src={product.media \|\| tshirtp}` + `onError` |
| `mixMatchActions.jsx` | `src={product.media}` | `src={product.media \|\| tshirtp}` + `onError` |
| `volumeDiscountActions.jsx` | `src={product.media}` | `src={product.media \|\| tshirtp}` + `onError` |
| `MixAndMatchEditor.jsx` | `src={product.media}` (×2) | `src={product.media \|\| tshirt}` + `onError` |
| `DiscountPreviewModal.jsx` | `src={product.media}` | `src={product.media \|\| PLACEHOLDER_IMG}` + `onError` |

## Changes

- Add `|| fallback` so `null`/`undefined`/empty media values display a placeholder immediately.
- Add `onError` handler to each image tag so broken remote URLs also fall back gracefully.
- New `DiscountPreviewModal.jsx` constant `PLACEHOLDER_IMG` (Shopify CDN placeholder) avoids a local asset import.
- 7 new unit tests in `tests/components/Modals/DiscountPreviewModal.test.jsx` covering all edge cases (null, undefined, empty string, broken URL, multi-product).

## Tests

```
Test Files  10 passed (10)
     Tests  97 passed (97)
```

Build: `CI=true npm run build` ✅ exit 0

_This PR was created by an AI agent (OpenHands) on behalf of the repository pipeline._

### CODE REVIEW

File: web/frontend/components/Modals/DiscountPreviewModal.jsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import { Modal, Button, Badge, Row, Col, ListGroup } from 'react-bootstrap';
 
+const PLACEHOLDER_IMG = 'https://cdn.shopify.com/s/files/1/0533/2089/files/placeholder-images-product-1_large.png?v=1530129292';
```
Comment (HeyItsChloe) [inline web/frontend/components/Modals/DiscountPreviewModal.jsx]: **Suggestion (non-blocking):** `PLACEHOLDER_IMG` points to an external Shopify CDN URL. Every other fixed file (bundleDiscountActions, mixMatchActions, volumeDiscountActions, MixAndMatchEditor) uses the locally-bundled `tshirt.png` asset, which is always available regardless of network conditions. Consider importing the same local asset here instead:

```jsx
import tshirtp from "../../assets/tshirt.png";
const PLACEHOLDER_IMG = tshirtp;
```
or simply inline it at the call-site as `product.media || tshirtp`. This makes the fallback consistent across all components and immune to external CDN downtime.

**Review by HeyItsChloe — COMMENTED** [review]
Automated review by OpenHands pipeline. One suggestion and two non-blocking observations.

File: web/frontend/components/Modals/DiscountPreviewModal.jsx (hunk 1 above)
Comment (HeyItsChloe) [reply web/frontend/components/Modals/DiscountPreviewModal.jsx]: Fixed in cee3fce — replaced the external Shopify CDN URL with `import tshirtp from "../../assets/tshirt.png"`, consistent with all other components. Test updated to import the same asset. 97/97 tests still green.
